pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -710,12 +713,32 @@ func (gce *GCECloud) EnsureLoadBalancer(clusterName string, apiService *api.Serv
 		glog.Infof("EnsureLoadBalancer(%v(%v)): deleted forwarding rule", loadBalancerName, serviceName)
 	}
 	if tpExists && tpNeedsUpdate {
-		if err := gce.deleteTargetPool(loadBalancerName, gce.region); err != nil {
+		hc := []*compute.HttpHealthCheck{}
```

Comment:
I don't understand why the healthcheck logic is in the targetpool conditional block.  Can you please comment the heck out of this?  This codepath is historically very brittle.